cmd/loki-curator/main.go

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+
+	"github.com/gardener/logging/cmd/loki-curator/app"
+	"github.com/gardener/logging/pkg/loki/curator"
+	"github.com/gardener/logging/pkg/loki/curator/metrics"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+func main() {
+	// metrics
+	go func() {
+		http.Handle("/curator/metrics", promhttp.Handler())
+		_ = http.ListenAndServe(":3101", nil)
+	}()
+
+	conf, logger, err := app.ParseConfiguration()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error: %v\n", err)
+		metrics.Errors.WithLabelValues(metrics.ErrorWithMainFunc).Inc()
```

Comment:
I doubt that anything will be able to scrape this metric because you immediately exit on the next line.
My point is that you can remove this line.